### ISSUE TITLE
pymmcore-plus integration

### ISSUE THREAD

**Issue by felixwanitschke** [jacopoabramo/napari-live-recording#15]
Adding the [pymmcore-plus](https://pymmcore-plus.github.io/pymmcore-plus/) camera device layer to be controlled via the plugin. The guidelines to integrate this layer are as follows:

- implementation of the interface described in the file `napari_live_recording/control/devices/interface.py`
- generalized access to all supported pymmcore-plus camera devices and settings
- testing the interface using the `DemoCamera` dummy interface provided by the package

**Comment by felixwanitschke** [comment]
Problem: Overfilled camera settings area, since there are in general too many camera properties that can be changed.

Idea: Including the main properties into the live-recording widget and outsource further settings to a seperate window.

But: Have to check, if there is a constistent approach how properties are implemented, how they are labeled etc.

https://github.com/micro-manager/mmCoreAndDevices/tree/main/MMCore/Devices

**Comment by jacopoabramo** [comment]
@felixwanitschke from the API what I noticed is that the only common way to access a property shared with all devices is the exposure time, because there is a common set of APIs that goes as follows:


``` C
void setExposure (double exp) throw (CMMError)
void setExposure (const char *cameraLabel, double dExp) throw (CMMError)
double getExposure () throw (CMMError)
double getExposure (const char *label) throw (CMMError)
```

This goes well with our case, and what you can do in the changeParameter function is to do something like this

```python
def changeParameter(name: str, value: Any) -> None:
    if name == "Exposure (ms)":
        self.__capture.setExposure(value)
        return
   ...
   # handle other parameters afterwards
```

Since for the MMCore the exposure is always expressed in milliseconds for cameras, this is valid for all devices and it's covered. If you can find anything which is similar to this it would be cool to add. If not we just add an extra button which opens up a separate dialog window.

**Comment by felixwanitschke** [comment]
Hello,

 

Okey, I did this.

 

Best Regards, 

Felix

**Comment by felixwanitschke** [comment]
`    def changeROI(self, newROI: ROI):
        self.setAcquisitionStatus(False)
        self.__capture.setROI(
            self.name, newROI.offset_x, newROI.offset_y, newROI.width, newROI.height
        )
        self.__capture.waitForDevice(self.name)
        self.setAcquisitionStatus(True)
        if newROI <= self.fullShape:
            self.roiShape = newROI`

- This is how I updated the change ROI method for pymmcore. I have several problems with that:

1. The first is that without the self.__capture.waitForDevice(self.name) the live view somehow stopped after changing the ROI, this was solved when toggling live once more or when chnaging the ROI again (to the same value). This was solved by the self.__capture.waitForDevice(self.name) 
2. When changing the ROI quickly (for example setting it to 10 in x direction and then quickly setting it to full-frame again) napari freezes
